cryptography/src/bls12381/primitives/group.rs

```diff
@@ -1045,4 +1081,196 @@ mod tests {
         let result_g2 = G2::msm(&points_g2, &scalars);
         assert_eq!(expected_g2, result_g2, "G2 MSM basic case failed");
     }
+
+    #[test]
+    fn test_scalar_ord() {
+        let mut rng = thread_rng();
```

Comment:
We should use a seeded rng to ensure this test is reproducible (there are a few places where this applies).

Reply:
If you wanted to go the "extra mile", we could loop through ~100 seeds to ensure these checks hold.